src/main/java/com/hillel/items_exchange/controller/UserController.java

```diff
@@ -0,0 +1,45 @@
+package com.hillel.items_exchange.controller;
+
+import com.hillel.items_exchange.dto.UserDto;
+import com.hillel.items_exchange.service.UserService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.MessageSource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import java.security.Principal;
+
+@RestController
+@RequestMapping("/user")
+@CrossOrigin(origins = "http://localhost:4200")
+@RequiredArgsConstructor
+@Slf4j
+public class UserController {
+
+    private final MessageSource messageSource;
+    private final UserService userService;
+
+    @GetMapping("/info/{id}")
+    public @ResponseBody
+    ResponseEntity<UserDto> getUserInfo(@Validated @PathVariable("id") Long id, Principal principal) {
+
+        UserDto userDto = userService.getByUsernameOrEmail(principal.getName()).orElse(new UserDto());
+        if (!id.equals(userDto.getId())){
```

Comment:
I can't understand why you have created a new instance of UserDto and then compare ids

Reply:
I did this to reduce the number of hits to the database. This way I already know the information about the user and if his id is equal to the input Id, then the user is given the information

Reply:
Maybe I'm mistaken, but you hit to the DB once taking User by name or nothing, so then you can work with that instance without any hitting to DB